src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.State.cs

```diff
@@ -46,6 +49,14 @@ private class State
 
                 public TrackedGeneratorDriver GeneratorDriver { get; }
 
+                /// <summary>
+                /// Weak table to the assembly symbols that this compilation tracker has created.  This can be used to
+                /// determine which project an assembly symbol came from after the fact.  This is needed as the compilation
+                /// an assembly came from can GC'ed and further requests to get that compilation (or any of it's assemblies)
+                /// may produce new assembly symbols.
+                /// </summary>
+                public readonly ConditionalWeakTable<ISymbol, ISymbol> CompilationAssembliesAndModules;
```

Comment:
>ConditionalWeakTable [](start = 32, length = 20)

What are the keys in this table?

Reply:
they are IAssemblySymbol or IModuleSymbols produced by `compilation.Assembly` or `compilation.GetAssemblyOrModule(MetadataReference)`.